bokehjs/src/coffee/common/embed.coffee

```diff
@@ -8,6 +8,18 @@ HasProperties = require "./has_properties"
 {pull_session} = require "./client"
 {Promise} = require "es6-promise"
 
+_handle_notebook_comms = (msg) ->
+  logger.debug("handling notebook comms")
+  # @ is bound to the doc
+  @apply_json_patch_string(msg.content.data)
```

Comment:
this assumes you have the PATCH-DOC type json not the PUSH-DOC type json. I think probably the json should include msgtype?

However, It looks like we don't have coffee code to handle PUSH-DOC right now, because we never send a PUSH-DOC from server to client on the websocket. That said. I think it's a one-liner:
`@replace_with_json(msg.content.data['doc'])`